### PR TITLE
Finish up resnet

### PR DESCRIPTION
**Pull request by mattdangerw**
- Add presets.
- Add converter script.
- Add preprocessing with auto resizing.

### CODE REVIEW

**Comment by mattdangerw** [comment]
Got some feedback from Francois on this offline. I'll tag reviewers again when I've addressed.

File: tools/checkpoint_conversion/convert_resnet_checkpoints.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    "resnet_v2_50": "timm/resnetv2_50.a1h_in1k",
+    "resnet_v2_101": "timm/resnetv2_101.a1h_in1k",
+}
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_string(
+    "preset",
+    None,
+    "Must be a valid `CausalLM` preset from KerasNLP",
```
Comment (sachinprasadhs) [inline tools/checkpoint_conversion/convert_resnet_checkpoints.py:56]: 'ResNet'

File: keras_nlp/src/models/resnet/resnet_presets.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ResNet preset configurations."""
+
+backbone_presets = {
+    "resnet_18": {
```
Comment (divyashreepathihalli) [inline keras_nlp/src/models/resnet/resnet_presets.py:17]: should we add v1 in the preset name?

**Review by divyashreepathihalli — APPROVED** [review]

File: keras_nlp/src/models/resnet/resnet_presets.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/src/models/resnet/resnet_presets.py:17]: We could. No strong opinion. Most models on HF forgo this, and only add it for v2.

Compare...
https://huggingface.co/models?sort=trending&search=resnet+v1
https://huggingface.co/models?sort=trending&search=resnet+v2

**Pull request merged**